tests/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Source.Tests/DiagnosticAnalysisRunnerTests.cs

```diff
@@ -0,0 +1,140 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Autofac.Extras.Moq;
+using AutoFixture;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.MSBuild;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers;
+using Microsoft.Extensions.Logging;
+using Moq;
+using Xunit;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Steps.Source.Tests
+{
+    public class DiagnosticAnalysisRunnerTests
+    {
+        private readonly Fixture _fixture;
+        internal const string TestProjectPath = @"assets\TestProject.csproj";
+
+        private static IEnumerable<AdditionalText> AdditionalTexts
+        {
+            get
+            {
+                return Array.Empty<AdditionalText>();
+            }
+        }
+
+        public DiagnosticAnalysisRunnerTests()
+        {
+            _fixture = new Fixture();
+        }
+
+        [Fact]
+        public async Task DiagnosticsWithEmptyProjectTest()
+        {
+            // Arrange
+            using var mock = AutoMock.GetLoose();
+            var loggerMock = new Mock<ILogger<DiagnosticAnalysisRunner>>();
+
+            var project = CreateProject(mock);
+            project.Setup(p => p.GetRoslynProject()).Returns(() =>
+            {
+                var ws = new AdhocWorkspace();
```

Comment:
I usually create these kind of things outside of the returns call. This *may be* fine, just verify that if it gets called twice, will it get the same one?

Reply:
Following @mjrousos's pattern here - https://github.com/dotnet/upgrade-assistant/blob/main/tests/extensions/web/Microsoft.DotNet.UpgradeAssistant.Steps.Razor.Tests/RazorSourceUpdaterTests.cs#L364

Reply:
Sure, but it's a pattern that I'm not sure it'll have the intended consequence. It may not matter here, but does it matter if the call is idempotent?